### ISSUE TITLE
[2.0.7] Form field empty with name "value" throws  BadMethodCallException

### ISSUE THREAD

**Issue by vikilaboy** [phalcon/cphalcon#10874]
If i have a form with empty values and a field with name "value" 

``` zephir
class AddressForm extends Form
{
    public function initialize()
    {
        $value = new Text('value', [
            'placeholder' => 'Street, number, block, ap. ...',
            'required'    => 'true',
            'autofocus'   => 'true'
        ]);
        $value->addValidators(
            [
                new PresenceOf([
                    'message' => 'Address is required.'
                ])
            ]
        );
        $this->add($value);
    }
}
```

``` php
<div class="col-sm-10">
    <?php echo $addressForm->render('value', ['class' => 'form-control input-sm']); ?>
</div>
```

on render i get BadMethodCallException: Wrong number of parameters

but if it's not empty the value of that field, it works fine

**Comment by andresgutierrez** [comment]
Could you please post the backtrace of the exception?

**Comment by vikilaboy** [comment]
![screen shot 2015-09-01 at 22 53 03](https://cloud.githubusercontent.com/assets/2757887/9614846/f189f794-50fc-11e5-8676-6ab059bb2a08.png)

**Comment by Green-Cat** [comment]
@vikilaboy What debugger is that? It looks pretty sweet.

**Comment by vikilaboy** [comment]
@Green-Cat - https://vimeo.com/68893840 :)

**Comment by jmsfwk** [comment]
I got a similar exception with `Text("label")`.

**Comment by andresgutierrez** [comment]
This must be fixed in 2.1.x, it tries to find if the form has a getter called "get"+ field, but the form class have both getValue and getLabel so any field having these names invoke the wrong getter. Renaming the convention in Phalcon 2.1.x

**Comment by vikilaboy** [comment]
Also in models which extends Phalcon\Mvc\Model; if you have a "source" column in DB will be confused with source table.

**Comment by dreamsxin** [comment]
@andresgutierrez All getter and setter add prefix `_`, how about?

**Comment by mapirelli** [comment]
There are news about this bug ?
I got the same error with Phalcon 2.0.8!
[06-Oct-2015 14:23:26 Europe/Berlin] PHP Fatal error:  Method Phalcon\Forms\Element\Text::__toString() must not throw an exception in /Users/mariano.pirelli/PHP/Momentum/app/classes/Momentum/Table/DSTableController.php on line 0

**Comment by andresgutierrez** [comment]
It will be fixed in Phalcon 2.1

**Comment by Jurigag** [comment]
Was it fixed?
